src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignAssembledContigsSpark.java

```diff
@@ -81,7 +73,31 @@ protected void runTool(final JavaSparkContext ctx) {
             }
 
         });
-        assembledBreakpoints.saveAsTextFile(output);
+        allContigAlignments.saveAsTextFile(output);
+    }
+
+    /**
+     * Loads an RDD of {@link ContigsCollection} objects keyed by assembly ID from disk. The input file
+     * should be the output of as RunSGAViaProcessBuilderOnSpark.
+     */
+    static JavaPairRDD<String, ContigsCollection> loadContigsCollectionKeyedByAssemblyId(final JavaSparkContext ctx, final String inputPath) {
+        final JavaRDD<String> inputAssemblies = ctx.textFile(inputPath).cache();
+
+        final JavaPairRDD<String, String> contigCollectionByBreakpointId =
+                inputAssemblies
+                        .flatMapToPair(RunSGAViaProcessBuilderOnSpark::splitAssemblyLine);
```

Comment:
Hmm. I've just been trying to keep the code that reads and parses data near the code that writes that data, so that you only have to change it one place. Maybe it needs to be separated out into a persistence class though, since both you and @SHuang-Broad have complained about it.